aws/resource_aws_athena_database.go

```diff
@@ -0,0 +1,208 @@
+		},
+	}
+
+	athenaresp, err := athenaconn.StartQueryExecution(athenainput)
+	if err != nil {
+		return err
+	}
+
+	if err := checkCreateDatabaseQueryExecution(*athenaresp.QueryExecutionId, d, meta); err != nil {
+		return err
+	}
+	d.SetId(d.Get("name").(string))
+	return resourceAwsAthenaDatabaseRead(d, meta)
+}
+
+func resourceAwsAthenaDatabaseRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).athenaconn
+
+	bucket := d.Get("bucket").(string)
+	input := &athena.StartQueryExecutionInput{
+		QueryString: aws.String(showDatabaseQueryString()),
+		ResultConfiguration: &athena.ResultConfiguration{
+			OutputLocation: aws.String("s3://" + bucket),
+		},
+	}
+
+	resp, err := conn.StartQueryExecution(input)
+	if err != nil {
+		return err
+	}
+
+	if err := checkShowDatabaseQueryExecution(*resp.QueryExecutionId, d, meta); err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceAwsAthenaDatabaseUpdate(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
+
+func resourceAwsAthenaDatabaseDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).athenaconn
+
+	bucket := d.Get("bucket").(string)
+	input := &athena.StartQueryExecutionInput{
+		QueryString: aws.String(dropDatabaseQueryString(d.Get("name").(string))),
+		ResultConfiguration: &athena.ResultConfiguration{
+			OutputLocation: aws.String("s3://" + bucket),
+		},
+	}
+
+	resp, err := conn.StartQueryExecution(input)
+	if err != nil {
+		return err
+	}
+
+	if err := checkDropDatabaseQueryExecution(*resp.QueryExecutionId, d, meta); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createDatabaseQueryString(databaseName string) string {
+	return fmt.Sprintf("create database %s;", databaseName)
+}
+
+func checkCreateDatabaseQueryExecution(qeid string, d *schema.ResourceData, meta interface{}) error {
+	rs, err := queryExecutionResult(qeid, meta)
+	if err != nil {
+		return err
+	}
+	if len(rs.Rows) != 0 {
+		return fmt.Errorf("[ERROR] Athena create database, unexpected query result: %s", flattenAthenaResultSet(rs))
+	}
+	return nil
+}
+
+func showDatabaseQueryString() string {
+	return fmt.Sprint("show databases;")
+}
+
+func checkShowDatabaseQueryExecution(qeid string, d *schema.ResourceData, meta interface{}) error {
+	rs, err := queryExecutionResult(qeid, meta)
+	if err != nil {
+		return err
+	}
+	found := false
+	dbName := d.Get("name").(string)
+	for _, row := range rs.Rows {
+		for _, datum := range row.Data {
+			if *datum.VarCharValue == dbName {
+				found = true
+			}
+		}
+	}
+	if !found {
+		return fmt.Errorf("[ERROR] Athena not found database: %s, query result: %s", dbName, flattenAthenaResultSet(rs))
+	}
+	return nil
+}
+
+func dropDatabaseQueryString(databaseName string) string {
+	return fmt.Sprintf("drop database %s;", databaseName)
+}
+
+func checkDropDatabaseQueryExecution(qeid string, d *schema.ResourceData, meta interface{}) error {
```

Comment:
I think better name for this function would be something like `executeAndExpectNoRows` or something like that - there's no logic related to database dropping (other than error message), so the name shouldn't be pretending there is.